woocommerce/wc-admin
# Add option to create product by template

Fixes woocommerce/woocommerce-admin#5296 

These changes will allow users to select a product template when creating their first product. The product template is generated from a CSV file very similar to that of the sample-data.csv, and using the same underlying code for importing.
Instead of publishing the product right away it is saved as `auto-draft`, similar to how a blank product is created.

This PR includes a small underlying refactor to the `<List />` package, to allow for custom list item content.
I also was not able to use the `<RadioControl />` component in the end, as it does not support subtitles.

### Accessibility


### Screenshots

https://user-images.githubusercontent.com/2240960/104055597-69c72c80-51c5-11eb-99f9-9b06d0de55c6.mp4

### Detailed test instructions:

- Load new store and finish the Wizard
- Go to the `Add my products` task
- Click the `Start with a template` option, and select either a physical, digital, variable product
- Once you click `Go`, it should redirect you to an edit page of the new post, with the data from the sample-data.csv (mentioned in the original ticket). Only the title is missing, as it is saved as auto-draft.
- You should be able to save the new product as draft or publish it.
- You should be able to exit without making any changes, and not having the product show up as draft in your product list. 
  - Create new product from template
  - Wait until redirected
  - Without saving go to the **Products > all products** page, the new product should not be displayed.
